source/broadband/cosa_webpa_dml.c

```diff
@@ -510,20 +579,150 @@ int getWebpaParameterValues(char **parameterNames, int paramCount, int *val_size
                     }
                     case 1:
                     {
-                        paramVal[k] = (parameterValStruct_t *) malloc(sizeof(parameterValStruct_t));
-                        if((strcmp(parameterNames[i], WEBPA_PARAM_VERSION) == 0) || ((isWildcard == 1) && (strcmp(parameterNames[i],webpaObjects[j]) == 0)))
-                        {
-                            paramVal[k]->parameterName = strndup(WEBPA_PARAM_VERSION, MAX_PARAMETERNAME_LEN);
-                            paramVal[k]->parameterValue = (char*) malloc(sizeof(char)*MAX_PARAMETERVALUE_LEN);
-                            snprintf(paramVal[k]->parameterValue,sizeof(char)*MAX_PARAMETERVALUE_LEN,"%s-%s",WEBPA_PROTOCOL, WEBPA_GIT_VERSION);
-                            paramVal[k]->type = ccsp_string;
-                            k++;
-                        }
-                        else
-                        {
-                            WAL_FREE(paramVal[k]);
-                            matchFound = 0;
-                        }
+			if(isWildcard == 0) {
+                            paramVal[k] = (parameterValStruct_t *) malloc(sizeof(parameterValStruct_t));
+                            if(strcmp(parameterNames[i], WEBPA_PARAM_VERSION) == 0) // Device.X_RDKCENTRAL-COM_Webpa.Version
+                            {
+                                paramVal[k]->parameterName = strndup(WEBPA_PARAM_VERSION, MAX_PARAMETERNAME_LEN);
+                                paramVal[k]->parameterValue = (char*) malloc(sizeof(char)*MAX_PARAMETERVALUE_LEN);
+                                snprintf(paramVal[k]->parameterValue,sizeof(char)*MAX_PARAMETERVALUE_LEN,"%s-%s",WEBPA_PROTOCOL, WEBPA_GIT_VERSION);
+                                paramVal[k]->type = ccsp_string;
+                                k++;
+                            }
+			    else if(strcmp(parameterNames[i], WEBPA_SERVER_URL) == 0)  //Device.X_RDKCENTRAL-COM_Webpa.Server.URL
+			    {			    
+                                paramVal[k]->parameterName = strndup(WEBPA_SERVER_URL, MAX_PARAMETERNAME_LEN);
```

Comment:
webpa server url get in lines 
594 - 600
646 - 652
686 - 693
is same. 
Please create one function and call it from these 3 places.
Same for the other 2 parameters too.

Reply:
This duplicates optimization is not done currently.